src/EditorFeatures/Core.Wpf/SignatureHelp/Controller_OnCaretPositionChanged.cs

```diff
@@ -17,6 +17,12 @@ internal override void OnCaretPositionChanged(object sender, EventArgs args)
         }
 
         private void OnCaretPositionChanged()
-            => Retrigger();
+        {
+            Retrigger(fromArrowKeys: _caretMoveFromArrowKeys);
+
+            // Always clear this flag because when argument snippet completion is active Tab characters
+            // will cause a caret move, but we don't want our special arrow key logic running.
+            _caretMoveFromArrowKeys = false;
```

Comment:
ok.  i see what you're doing, and i don't like it.  i have an alternate suggestion, though i don't know if ti would work or not.  Instead of having the OnCaretPositionChanged read this value, instead just have the arrow handling helpers above directly call Retrigger.  Then (i'm hoping) we don't need OnCarePositionChanged anyways, becaue we'd directly handling all the ways the caret can move.

Reply:
Unfortunately we can't retrigger here because the caret hasn't moved yet. We need to let the editor do its stuff to work out where the caret needs to go. Also we still need OnCaretPositionChanged for the other reasons the caret can move.

Reply:
> Unfortunately we can't retrigger here because the caret hasn't moved yet
can we not call nextHandler() to pass things along so that the caret goes through, and then we retrigger *after* That goes through?

Reply:
Firstly, thank you, I would have never thought of that. I figured the caret movement would be done after command handlers, not as a command handler.

Unfortunately if we let the editor move the caret first, then we need to know to ignore the next CaretPositionChanged event, which we'd need state to track.

> directly handling all the ways the caret can move.

I guess we could do that, but I feel like that's going a bit far to avoid one bool on a class that already stores some state anyway. For example, 6 months ago Tab and Shift+Tab didn't cause a caret position change, but now because of snippet parameter completion they do. I don't even know what the complete list of ways the caret can move are, let alone want to try to keep up with the editor in maintaining support for them all.